pkg/controller/operators/catalog/operator.go

```diff
@@ -1654,11 +1667,62 @@ func (o *Operator) ExecutePlan(plan *v1alpha1.InstallPlan) error {
 				plan.Status.Plan[i].Status = status
 
 			default:
-				return v1alpha1.ErrInvalidInstallPlan
-			}
+				if !isSupported(step.Resource.Kind) {
```

Comment:
This section looks good, but I think we should do some refactoring now that we have this.

Plan execution can be much simpler, now, roughly:

```
 if kind == CSV { create, continue }
 if kind == CRD  && CRD doesn't exist { create, requeue }
 if kind == CRD  && CRD exists && CRD not ready { requeue }
 if isSupported(kind) { treat as unstructured, create, continue }
```

in other words, we need to special case CRD/CSV handling for ordering reasons, but everything else that we support can be treated the same (for the most part; I think serviceaccounts might need some special handling as well).

Reply:
@ecordell and I discussed this offline and I will follow up this PR with a PR that refactors this code.